app/controllers/responses_controller.rb

```diff
@@ -68,6 +74,21 @@ def response_params
       params.require(:response).permit(answers_attributes: [{raw: []}, :raw, :answerable_type, :answerable_id])
     end
 
+    def validate_responses(params)
+      answers = params["answers_attributes"].values
+
+      required_questions = Question.pluck(:id)
+      required_demographic_questions = DemographicQuestion.where(validation: {required: true}.to_json).pluck(:id)
```

Comment:
Would make sense to move these to the actual objects so you could simply do `Question.required_questions` or something similar for both

Reply:
Done now!